samples/VanillaJSTestApp2.0/app/default/test/browser.spec.ts

```diff
@@ -0,0 +1,120 @@
+import "mocha";
+import puppeteer from "puppeteer";
+import { expect } from "chai";
+import fs from "fs";
+import { TestCredential } from "../../../e2eTests/TestCredential";
+
+const SCREENSHOT_BASE_FOLDER_NAME = `${__dirname}/screenshots`;
+let SCREENSHOT_NUM = 0;
+let username = "";
+let accountPwd = "";
+
+function setupScreenshotDir() {
+    if (!fs.existsSync(`${SCREENSHOT_BASE_FOLDER_NAME}`)) {
+        fs.mkdirSync(SCREENSHOT_BASE_FOLDER_NAME);
+    }
+}
+
+async function setupCredentials() {
+    const testCreds = new TestCredential();
+    const envResponse = await testCreds.getUserVarsByCloudEnvironment("azureppe");
+    const testEnv = envResponse[0];
+    if (testEnv.upn) {
+        username = testEnv.upn;
+    }
+
+    const testPwdSecret = await testCreds.getSecret(testEnv.labName);
+
+    accountPwd = testPwdSecret.value;
+}
+
+async function takeScreenshot(page: puppeteer.Page, testName: string, screenshotName: string): Promise<void> {
+    const screenshotFolderName = `${SCREENSHOT_BASE_FOLDER_NAME}/${testName}`
+    if (!fs.existsSync(`${screenshotFolderName}`)) {
+        fs.mkdirSync(screenshotFolderName);
+    }
+    await page.screenshot({ path: `${screenshotFolderName}/${++SCREENSHOT_NUM}_${screenshotName}.png` });
+}
+
+async function enterCredentials(page: puppeteer.Page, testName: string): Promise<void> {
+    await page.waitForNavigation({ waitUntil: "networkidle0"});
+    await takeScreenshot(page, testName, `loginPage`);
+    await page.type("#i0116", username);
+    await page.click("#idSIButton9");
+    await page.waitForNavigation({ waitUntil: "networkidle0"});
+    await takeScreenshot(page, testName, `pwdInputPage`);
+    await page.type("#i0118", accountPwd);
+    await page.click("#idSIButton9");
```

Comment:
should all of these strings be constants?

Reply:
I will enhance some of this in a future PR.

Reply:
+1